packages/core/src/renderer/components/+workloads-replicationcontrollers/replicationcontroller-details.tsx

```diff
@@ -5,24 +5,114 @@
 
 import styles from "./replicationcontroller-details.module.scss";
 import React from "react";
+import { makeObservable, observable } from "mobx";
 import { observer } from "mobx-react";
 import { withInjectables } from "@ogre-tools/injectable-react";
 import { DrawerItem, DrawerTitle } from "../drawer";
 import { Badge } from "../badge";
 import type { KubeObjectDetailsProps } from "../kube-object-details";
 import type { ReplicationControllerStore } from "./replicationcontroller-store";
 import replicationControllerStoreInjectable from "./replicationcontroller-store.injectable";
-import type { ReplicationController } from "../../../common/k8s-api/endpoints";
+import type {
+  ReplicationController,
+  ReplicationControllerApi,
+} from "../../../common/k8s-api/endpoints";
+import replicationControllerApiInjectable
+  from "../../../common/k8s-api/endpoints/replication-controller.api.injectable";
+import { Button } from "../button";
+import { Input } from "../input";
+import showErrorNotificationInjectable from "../notifications/show-error-notification.injectable";
+import type { ShowNotification } from "../notifications";
+import hideDetailsInjectable from "../kube-detail-params/hide-details.injectable";
 
 export interface ReplicationControllerDetailsProps extends KubeObjectDetailsProps<ReplicationController> {
 }
 
 interface Dependencies {
   store: ReplicationControllerStore;
+  api: ReplicationControllerApi;
+  showNotificationError: ShowNotification;
+
+  hideDetails(): void;
 }
 
 @observer
-class NonInjectedReplicationControllerDetails extends React.Component<ReplicationControllerDetailsProps & Dependencies> {
+class NonInjectedReplicationControllerDetails<Props extends ReplicationControllerDetailsProps & Dependencies> extends React.Component<Props> {
+  @observable showScaleDialog = false;
+
+  constructor(props: Props) {
+    super(props);
+    makeObservable(this);
+  }
+
+  inputRef = React.createRef<Input>();
+
+  scale(replicas: number) {
+    const { object: resource, api } = this.props;
+
+    return api.scale({
+      name: resource.getName(),
+      namespace: resource.getNs(),
+    }, replicas);
+  }
+
+  renderReplicasAndScaleDialog() {
+    const { object: resource, showNotificationError, hideDetails } = this.props;
+
+    if (this.showScaleDialog) {
```

Comment:
I think it would be a better UX to have a slider + scale button instead of a text input field

Reply:
Ok, I'll try to play around with slider and see how does it look & feel.

Reply:
Updated with using Slider component, PTAL